### DISCUSSION TITLE
Delayed Execution of Scheduled Cron Job

### DISCUSSION THREAD

**Discussion by mehditareghi** [vercel/community#3424] [Help] [answered]
### Summary

I hope this message finds you well. I am facing an issue with a scheduled cron job on my Vercel project, and I am seeking community input and support to resolve it.

Problem Description:
The cron job is designed to execute the "introMeeting.js" function at specific intervals for sending scheduled emails to our users. While the function itself works correctly and has been successfully tested on another "cron.js" file within the same directory, there seems to be a significant delay in the execution of this particular cron job.

Logs from Vercel:

    Time: July 30, 01:11:28.20 GMT+03:30
    Status Code: 200
    Host: dashboards-jvbmztp0b-darsoon.vercel.app
    Request ID: szjxc-1690666888188-8262b393b895
    Request User Agent: vercel-cron/1.0
    Level: Info
    Type: Edge Middleware
    Function: /middleware
    [GET] [middleware: "middleware"] /api/introMeeting.js status=200

Expected Outcome:
Following this log, I would normally expect additional logs indicating that the function has run successfully or an error log in case of any issues during execution. However, the logs seem to end at this point, and the scheduled emails are not being sent promptly.

Troubleshooting Attempted:

    The "introMeeting.js" file has been reviewed and confirmed to be functioning correctly when executed manually.
    The "vercel.json" file contains the correct schedule for the cron job and has been tested successfully after a delay of about one day.
    Checked for possible causes of the delay, including server load, environment variables, timezone, and clock synchronization.

Request for Community Input:
I am posting this message to the GitHub community to seek input and advice from fellow developers who might have encountered similar issues or have expertise in cron jobs and Vercel deployments.

If you have any insights, suggestions, or troubleshooting steps that could help me identify the root cause of the delay and ensure that the cron job executes as expected at the specified intervals, I would greatly appreciate your contribution.

### Steps to Reproduce

To replicate the delayed execution issue of the scheduled cron job, follow the steps below:

    Configure the cron job:
        Set up a Vercel project with the "introMeeting.js" file in the api directory of your Next.js project.
        Ensure the "vercel.json" file specifies the desired schedule for the cron job.

    Monitor the logs:
        Observe the Vercel logs while the cron job is scheduled to run.

    Expected Behavior:
        Normally, after the cron job is triggered at the specified time, the logs should show the execution of the "introMeeting.js" function, followed by logs indicating the successful sending of emails and relevant database entries.

**Comment by amyegan** [comment]
Hey @mehditareghi. Thanks for reporting this. I have a couple of question to help point us in the right direction.

- Does it ever run on time, or is it always late? How long is the delay?
- About how long does the function take to complete when run manually?
- Would you be able to share your cron config here with sensitive values censored?

**Reply by mehditareghi** [reply]
I even created an extremely simple 'simpleLog.js' file in my api folder to test if the problem is on my side or not...

This is the content of this file:

console.log("something");

And in vercel.json I have this:

{

  "crons": [

    {

      "path": "/api/cron.js",

      "schedule": "*/2 * * * *"

    },

    {

      "path": "/api/introMeeting.js",

      "schedule": "*/2 * * * *"

    },

    {

      "path": "/api/simpleLog.js",

      "schedule": "*/1 * * * *"

    }

  ]

}

Even for simpleLog.js, I get just a status code: 200, and not get the log in the console!

Please care about this problem and provide me some instructions.

**Comment by mehditareghi** [comment]
Hello @amyegan. Let me explain the problem I'm encountering in more detail. The issue doesn't lie in the cron job starting late, instead, it revolves around a strange behavior when I generate a new file for a cron job in my API folder.

When I initially create a new cron job file, the job doesn't immediately begin executing its function as expected. Instead, the job merely sends a status code in response to the request. This initial period lasts for a few days and during this time, the function linked to the cron job doesn't run at all, it only returns the status code.

However, after a few days, the cron job finally starts executing the function as it should. What's more interesting is that once the function starts running correctly, I'm able to modify the existing cron job or even insert a new one into that same file. Once I do this, the jobs work perfectly, adhering to the designated schedule and performing their functions without any issues or delays.

Yet, when I create a new file with identical functionality, the issue resurfaces. The newly created cron job again takes a few days to start running the function and only sends status codes during that period.

Given this pattern, I'm inclined to believe that the cron configuration isn't the root cause of the problem, as once a file's cron job starts running the function, all subsequent jobs in that file run flawlessly. It seems that the delay only occurs when I set up a cron job in a brand new file.

**Reply by amyegan** [reply]
I'm still not able to replicate this issue. Would you be able to share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) or video showing what you did to add the new cron job and what result you see?

**Reply by David-orovs** [reply]
Hi @amyegan please help here https://github.com/orgs/vercel/discussions/4663

**Comment by David-orovs** [comment]
Hi @mehditareghi was this ever resolved cos i am having the same issue, see link to issue here https://github.com/orgs/vercel/discussions/4663

**Reply by mehditareghi** [reply, marked as answer]
Hi,
That was my silly mistake...
In vercel.json file, I put "api/example.js" for route...
After removing .js part, it worked.

**Reply by David-orovs** [reply]
hmm alright thanks
